modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/plugin/json/GenericJsonDescriptorProperty.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.deployment.internal.plugin.json;
+
+import org.mule.runtime.api.meta.model.ExtensionModel;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+/**
+ * Generic descriptor that will be used to describe parametrization to construct {@link ExtensionModel},
+ * {@link ClassLoader} and any other descriptor that may arise in a future of {@link MulePluginJsonDescriber}.
+ * <p/>
+ * Each {@link GenericJsonDescriptorProperty} will have an ID that will be used to discover any loader that's responsible of working
+ * with the current set of attributes. It's up to each loader to validate the types, size and all that matters around
+ * the attributes.
+ *
+ * @since 4.0
+ */
+public class GenericJsonDescriptorProperty {
```

Comment:
No. This class, `GenericJsonDescriptorProperty` is meant to be used only to read the JSON file to then be discarded. 
It's used in the `ArtifactPluginDescriptorFactory` (see `org.mule.runtime.module.deployment.internal.plugin.ArtifactPluginDescriptorFactory#lookupJsonDescriptor`)